tasks/Task.cpp

```diff
@@ -26,178 +26,128 @@ Task::~Task()
-    ret = BVTColorMapper_MapImage(mapper, img, &cimg);
-    if( ret != 0 )
-    {
-        printf("BVTColorMapper_MapImage: ret=%d\n", ret);
-        return 1;
+        first_ping = true;
     }
-    printf("\n");
-
-    /////////////////////////////////////////////////////////
-    // Check the image height and width out
-    BVTColorImage_GetHeight(cimg, &height);
-    printf("BVTColorImage_GetHeight: %d\n", height);
-    BVTColorImage_GetWidth(cimg, &width);
-    printf("BVTColorImage_GetWidth: %d\n", width);
-
-
-    // Save it to a PPM (PortablePixMap)
-    ret = BVTColorImage_SavePPM(cimg, "cimg.ppm");
-    if( ret != 0 )
+    catch (BVTSDK::SdkException &e)
     {
-        printf("BVTColorImage_SavePPM: ret=%d\n", ret);
-        return 1;
+        std::cout << e.ErrorName() << std::endl;
+        throw std::runtime_error(e.ErrorMessage());
     }
 
-    // Clean up
-    BVTColorImage_Destroy(cimg);
-    BVTMagImage_Destroy(img);
-    BVTColorMapper_Destroy(mapper);
-    BVTPing_Destroy(ping);
-    BVTSonar_Destroy(son);
     return true;
+    
 }
 bool Task::startHook()
 {
     if (! TaskBase::startHook())
+    {
         return false;
+    }
     return true;
 }
 void Task::updateHook()
 {
     TaskBase::updateHook();
+
+    try
+    {
+        //Get Ping
+        BVTSDK::Ping ping(head.GetPing(-1));
+
+        // Save ping in the file
+        if(_save_file.get())
+        {
+            head_file.PutPing(ping);
+        }
+
+        // Generate an image from the ping
+        BVTSDK::ImageGenerator img_gen;
+        img_gen.SetHead(head);
+
+        // Get the R-Theta image from the ping
+        BVTSDK::MagImage img_rtheta(img_gen.GetImageRTheta(ping));
+
+        // Set SonarScan parameters
+        if (first_ping)
+        {
+            this->setSonarParameters(ping, img_rtheta);
+            first_ping = false;
+        }
+
+        // Get the raw data pointer
+        unsigned short *bit_buffer = img_rtheta.GetBits();
+        bvt_sonar.bins.clear();
+
+        bvt_sonar.bin_count = img_rtheta.GetHeight();
+        int beams = bvt_sonar.beam_count;
+        int bins = bvt_sonar.bin_count;
+        bvt_sonar.bins.clear();
+        for (int i = 0; i < beams ; ++i)
+        {
+            for (int j = bins; j > 0 ; --j )
+            {
+                float value = (bit_buffer[(j*beams+i)]*1.0);
+                bvt_sonar.bins.push_back(value);
+            }
+        }
+
+        bvt_sonar.time = base::Time::now();
+        _sonar_output.write(bvt_sonar);
+
+        base::samples::SonarScan bvt_scan;
+        bvt_scan = convertSonarToSonarScan(bvt_sonar);
+        _sonar_scan.write(bvt_scan);
+
+        float temperature;
+        if (son.TryGetTemperature(&temperature))
+        {
+            _sonar_temperature.write((double)son.GetTemperature());
+        }
+    }
+    catch (BVTSDK::SdkException &e)
+    {
+        std::cout << e.ErrorName() << std::endl;
+        throw std::runtime_error(e.ErrorMessage());
```

Comment:
This was the way that I found to print the error message on the screen. Let the exception pass through just breaks the code and doesn't inform to the user what happened.
Is there another way to do that?